skimage/color/colorlabel.py

```diff
@@ -195,34 +196,110 @@ def _label2rgb_overlay(label, image=None, colors=None, alpha=0.3,
     return result
 
 
-def _label2rgb_avg(label_field, image, bg_label=0, bg_color=(0, 0, 0)):
+def _apply_per_channel(func, image):
+    """
+    Apply a function to each channel of an image.
+
+    Parameters
+    ----------
+    func : callable
+        A function that takes an array to another array of the same shape.
+    image : array
+        A single or multichannel image.
+
+    Returns
+    -------
+    out : array, same shape and type as `image`
+    """
+    if len(image.shape) not in [2, 3]:
+        raise ValueError('Image must be 2d or 3d array')
+    if len(image.shape) == 2:
+        return func(image)
+    else:
+        out = np.zeros(image.shape)
+        n = image.shape[2]
+        for k in range(n):
+            out[..., k] = func(image[..., k])
+        return out
+
+
+def _get_means_from_contiguous_regions(label_field, image):
+    """
+    Aggregates pixel values with respect to
+    labels for an image assuming that label_field
+    has the same shape as image. In particular,
+    labels will be averaged over all axes.
+
+    Parameters
+    ----------
+    label_field : array
+        2d or 3d array of labels (0, 1, ..., n) that indicate contiguous
+        regions to average
+    image : 2d or 3d array with shape `label_field.shape`
+
+    Returns
+    -------
+    out : array
+        An array with the same shape and type as `image` and `label_field`
+    """
+    if label_field.shape != image.shape:
+        raise ValueError('label_field and image must have the same shape as image')
+
+    # scipy wants labels to begin at 1 and transforms labels to 1, 2, ..., n + 1
+    labels_ = label_field + 1
+    labels_unique = np.unique(labels_)
+
+    out = np.zeros(image.shape)
+    indices = ndi.find_objects(labels_)
+
+    means = ndi.measurements.mean(image, labels=labels_, index=labels_unique)
+    for label, mean in zip(labels_unique, means):
+        indices_temp = indices[label - 1]
+        out[indices_temp][labels_[indices_temp] == label] = mean
+    return out
+
+
+def _label2mean(label_field, image, bg_label=None, bg_color=None, multichannel=True):
```

Comment:
```suggestion
def _label2mean(label_field, image, bg_label=None, bg_color=None, channel_axis=-1):
```
It would be great to update this parameter!

References:
* https://github.com/scikit-image/scikit-image/blob/main/doc/release/release_0.19.rst#announcement-scikit-image-0190
* https://scikit-image.org/docs/stable/user_guide/numpy_images.html?highlight=channel_axis#color-images
* https://github.com/scikit-image/scikit-image/blob/b566789687b2ff575fa21189d3a8d7b5fa0af287/skimage/color/colorconv.py#L63